src/charm.py

```diff
@@ -200,10 +200,23 @@ def change_config(self, charm_state: CharmState) -> None:
             return
         self.model.unit.status = ops.MaintenanceStatus("Configuring Synapse")
         try:
+            signing_key_path = f"/data/{charm_state.synapse_config.server_name}.signing.key"
```

Comment:
Is this needed for a single unit deployment?

Reply:
No, if there is just one unit it wont get the signing keys from a secret but it will be necessary to copy the existing signing key to a secret in case of a scaling.